# Business Understanding

Your company now sees all the big companies creating original video content and they want to get in on the fun. They have decided to create a new movie studio, but they don’t know anything about creating movies. You are charged with exploring what types of films are currently doing the best at the box office. You must then translate those findings into actionable insights that the head of your company's new movie studio can use to help decide what type of films to create.

# Data Understanding

The data for the solving the business problem comes from the following sources: Box Office Mojo (https://www.boxofficemojo.com/), IMDB (https://www.imdb.com/), Rotten Tomatos (https://www.rottentomatoes.com/), The Movie Database (https://www.themoviedb.org/?language=en-US), and The Numbers (https://www.the-numbers.com/).

Before solving for the business problem, I want to examine data from the aforementioned sources.  When I conduct exploratory data analysis of data, I aim to understand the following attributes:

1. Number of columns
2. Number of rows
3. Columns Names

In [4]:
# Importing libraries

import itertools
import numpy as np
import pandas as pd 
from numbers import Number
import sqlite3
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

import pickle

## IMDB | Data Understanding

The dataset for IMDB is contained within a zip file.  Before conducting exploratory data analysis, I will need to unzip the file.  Subsequently, I will utilize the following method - 'pd.read_sql' - to open any files and undestand the associated tables.

In [2]:
# # Opening the "im.db.zip" file and understand the name(s) of the files within the "im.db.zip" file

# import zipfile

# with zipfile.ZipFile('im.db.zip', 'r') as zip_ref:
#     zip_ref.extractall('movie_analysis_phase_2') 
#     file_list = zip_ref.namelist()

# print(file_list)

['im.db']


In [1]:
! ls

README.md
data
movie_analysis_notebook.ipynb


In [2]:
# Checking to ensure that the path for the "im.db.zip" file is correct

import os

print(os.path.exists('data/im.db'))

True


In [5]:
# Establishing the database connection path for the "im.db.zip" file

conn = sqlite3.connect('data/im.db')
cursor = conn.cursor()

In [7]:
# Executing the query to examine all of the tables in the "im.db" file

cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
print(cursor.fetchall())

[('movie_basics',), ('directors',), ('known_for',), ('movie_akas',), ('movie_ratings',), ('persons',), ('principals',), ('writers',)]


In [8]:
# Examining the contents of the "movie_basics" table

query_movie_basics = """SELECT * 
        FROM movie_basics;"""
        
pd.read_sql(query_movie_basics, conn)

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

In [7]:
# Examining the contents of the "principals" table

query_principals = """SELECT * 
        FROM principals;"""
        
pd.read_sql(query_principals, conn)

movie_id  ordering   person_id  category       job  \
0        tt0111414         1   nm0246005     actor      None   
1        tt0111414         2   nm0398271  director      None   
2        tt0111414         3   nm3739909  producer  producer   
3        tt0323808        10   nm0059247    editor      None   
4        tt0323808         1   nm3579312   actress      None   
...            ...       ...         ...       ...       ...   
1028181  tt9692684         1   nm0186469     actor      None   
1028182  tt9692684         2   nm4929530      self      None   
1028183  tt9692684         3  nm10441594  director      None   
1028184  tt9692684         4   nm6009913    writer    writer   
1028185  tt9692684         5  nm10441595  producer  producer   

                   characters  
0                 ["The Man"]  
1                        None  
2                        None  
3                        None  
4            ["Beth Boothby"]  
...                       ...  
1028181  ["Ebenezer Scrooge"]  
1028182   ["Herself","Regan"]  
1028183                  None  
1028184                  None  
1028185                  None  

[1028186 rows x 6 columns]

In [8]:
# Examining the contents of the "persons" table

query_persons = """SELECT * 
        FROM persons;"""
        
pd.read_sql(query_persons, conn)

person_id         primary_name  birth_year  death_year  \
0       nm0061671    Mary Ellen Bauder         NaN         NaN   
1       nm0061865         Joseph Bauer         NaN         NaN   
2       nm0062070           Bruce Baum         NaN         NaN   
3       nm0062195         Axel Baumann         NaN         NaN   
4       nm0062798          Pete Baxter         NaN         NaN   
...           ...                  ...         ...         ...   
606643  nm9990381         Susan Grobes         NaN         NaN   
606644  nm9990690          Joo Yeon So         NaN         NaN   
606645  nm9991320       Madeline Smith         NaN         NaN   
606646  nm9991786  Michelle Modigliani         NaN         NaN   
606647  nm9993380       Pegasus Envoyé         NaN         NaN   

                                      primary_profession  
0              miscellaneous,production_manager,producer  
1             composer,music_department,sound_department  
2                             miscellaneous,actor,writer  
3       camera_department,cinematographer,art_department  
4       production_designer,art_department,set_decorator  
...                                                  ...  
606643                                           actress  
606644                                           actress  
606645                                           actress  
606646                                          producer  
606647                             director,actor,writer  

[606648 rows x 5 columns]

In [9]:
# Placing the "movie_basics" table in a dataframe

query_movie_basics = """SELECT * 
        FROM movie_basics;"""
        
df_imdb_movie_basics = pd.read_sql(query_movie_basics, conn)

In [10]:
# Closing the connection to the file since I have already made the desired dataframes

conn.close()

In [11]:
# Examining the top 5 rows of the "df_imdb_movie_basics" dataframe

df_imdb_movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [12]:
# Examining the bottom 5 rows of the "df_imdb_movie_basics" dataframe

df_imdb_movie_basics.tail()

movie_id                                primary_title  \
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes       genres  
146139            123.0        Drama  
146140              NaN  Documentary  
146141              NaN       Comedy  
146142            116.0         None  
146143              NaN  Documentary

In [13]:
# Utilizing the ".info()" method to further understand the "df_imdb_movie_basics" dataframe

df_imdb_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [14]:
# Examining the shape of the "df_imdb_movie_basics" dataframe

df_imdb_movie_basics.shape

(146144, 6)

In [15]:
# Determining whether or not there are any duplicate rows in the "df_imdb_movie_basics" dataframe

imdb_movie_basics_duplicates = df_imdb_movie_basics[df_imdb_movie_basics.duplicated()]
print(len(imdb_movie_basics_duplicates))

0


In [16]:
# Determining whether or not there are any rows with null values within the "df_imdb_movie_basics" dataframe

df_imdb_movie_basics.isna().sum()

movie_id               0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

### IMDB | Data Understanding (Conclusion)

I have completed the exploratory data analysis of the IMDB zip file. Within the zip file, there was only one file, which was titled "im.db". Upon unzipping the file, I utilized the following code - 'cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")' - to understand the various tables.

The tables within the im.db file are the following: *movie_basics*, *directors*, *known_for*, *movie_akas*, *movie_ratings*, *persons*, *principals*, *writers*. This is congruent with the entity relationship diagram (ERD). Upon ascertaining the various tables, I explored the different columns within each table.

I decided to place the *movie_basics* table into a dataframe, which is titled *df_imdb_movie_basics*. The tables has a total of 146,144 rows and 6 columns. The movie_basics table has a column - *runtime_minutes* - that I am interested in using for analytics. However, there are 31,739 cells within the runtime_minutes column that has null values. I may have to delete the rows to further my eventual analysis.

## Box Office Mojo | Data Understanding

The dataset for Box Office Mojo is contained within a csv file.  I will utilize the following method - 'pd.read_csv()' - to examine the contents of the csv file.

In [17]:
# Reading the contents of the Box Office Mojo csv to look at the columns and content

pd.read_csv('bom.movie_gross.csv.gz')

title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

In [18]:
# Placing the Box Office Mojo file into a Dataframe

df_bom = pd.read_csv('bom.movie_gross.csv.gz')

In [19]:
# Examining the first five rows of the Box Office Mojo dataframe

df_bom.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [20]:
# Examining the last five rows of the Box Office Mojo dataframe

df_bom.tail()

title      studio  domestic_gross foreign_gross  \
3382                    The Quake       Magn.          6200.0           NaN   
3383  Edward II (2018 re-release)          FM          4800.0           NaN   
3384                     El Pacto        Sony          2500.0           NaN   
3385                     The Swan  Synergetic          2400.0           NaN   
3386            An Actor Prepares       Grav.          1700.0           NaN   

      year  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [21]:
# Utilizing the ".info()" method to further understand the Box Office Mojo dataframe

df_bom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [22]:
# Examining the shape of the Box Office Mojo dataframe

df_bom.shape

(3387, 5)

In [23]:
# Determining whether or not there are any duplicate rows in the Box Office Mojo dataframe

bom_duplicates = df_bom[df_bom.duplicated()]
print(len(bom_duplicates))

0


In [24]:
# Determining whether or not there are any rows with null values in the Box Office Mojo dataframe

df_bom.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

### Box Office Mojo | Data Understanding (Conclusion)

I have completed the exploratory data analysis of the csv file associated with Box Office Mojo.  I placed the contents of the csv file into a dataframe titled *df_bom*.  *Df_bom* has the following five columns - *title*, *studio*, *domestic_gross*, *foreign_gross*, and *year*.  *Df_bom* has a total of 3,387 rows. 

## Rotten Tomatoes (Movie Information) | Data Understanding

One of the datasets for Rotten Tomatoes is contained within a tsv file.  The data within the tsv file provides movie information.  I will utilize the following method - 'pd.read_csv()' - to examine the contents of the tsv file. 

In [26]:
# Placing the Rotten Tomatos (Movie Information) file into a Dataframe

df_rotten_movie = pd.read_csv('rt.movie_info.tsv.gz', sep='\t')

In [27]:
# Examining the first five rows of the Rotten Tomatos (Movie Information) dataframe

df_rotten_movie.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [28]:
# Examining the last five rows of the Rotten Tomatos (Movie Information) dataframe

df_rotten_movie.tail()

id                                           synopsis rating  \
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures

In [29]:
# Utilizing the ".info()" method to further understand the "df_rotten_movie" dataframe

df_rotten_movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [30]:
# Utilizing the ".nunique()" method to further understand the number of unique values for the 'studio' column 

df_rotten_movie['studio'].nunique()

200

In [31]:
# Utilizing the ".unique()" method to understand the various studios populated in the 'studio' column

df_rotten_movie['studio'].unique()[0:20]

array([nan, 'Entertainment One', 'Warner Bros. Pictures',
       'Paramount Pictures', 'Sony Pictures Classics',
       'Showtime Documentary Films', 'Seventh Art Releasing',
       'ATO Pictures', 'Sony Pictures', 'Universal Pictures', 'MGM',
       'After Dark Films/Freestyle Releasing', 'Lions Gate Films',
       'Regent Releasing', 'Janus Films', 'The Weinstein Company',
       'New Line Cinema', 'Walt Disney Pictures', 'FilmDistrict',
       '20th Century Fox'], dtype=object)

In [32]:
# Utilizing the ".value_counts" method to understand which studios are the most populous in the "df_rotten_movie" dataframe

df_rotten_movie['studio'].value_counts()[0:20]

Universal Pictures          35
Paramount Pictures          27
20th Century Fox            26
Sony Pictures Classics      22
Warner Bros. Pictures       21
Sony Pictures               16
Columbia Pictures           10
New Line Cinema             10
IFC Films                    9
Miramax Films                8
Warner Bros.                 8
MGM                          7
Magnolia Pictures            7
Fox Searchlight Pictures     7
Focus Features               6
Roadside Attractions         6
Walt Disney Pictures         6
Lionsgate Films              6
Lions Gate Films             6
The Weinstein Company        6
Name: studio, dtype: int64

In [33]:
# Utilizing the ".nunique()" method to further understand the number of unique values for the 'writer' column

df_rotten_movie['writer'].nunique()

1069

In [35]:
# Utilizing the ".unique()" method to understand the various writers populated in the 'writer' column

df_rotten_movie['writer'].unique()

array(['Ernest Tidyman', 'David Cronenberg|Don DeLillo', 'Allison Anders',
       ..., 'Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner',
       'David Mickey Evans|Robert Gunter', 'Luc Besson'], dtype=object)

In [34]:
# Utilizing the ".value_counts" method to understand which writers are the most populous in the 'writer' column

df_rotten_movie['writer'].value_counts()

Woody Allen                                    4
Jim Jarmusch                                   3
John Hughes                                    3
Sylvester Stallone                             3
Hong Sang-soo                                  3
                                              ..
Leslie Cockburn|Andrew Cockburn                1
Mario Puzo|David Newman|Leslie Newman          1
Mark Monroe                                    1
Yoji Yamada|Ichiro Yamamoto|Emiko Hiramatsu    1
Alex Ross Perry                                1
Name: writer, Length: 1069, dtype: int64

In [36]:
# Examining the shape of the "df_rotten_movie" dataframe

df_rotten_movie.shape

(1560, 12)

In [37]:
# Determining whether or not there are any duplicate rows in the "df_rotten_movie" dataframe

rotten_movie_duplicates = df_rotten_movie[df_rotten_movie.duplicated()]
print(len(rotten_movie_duplicates))

0


In [38]:
# Determining whether or not there are any rows with null values in the "df_rotten_movie" dataframe

df_rotten_movie.isna().sum()

id                 0
synopsis          62
rating             3
genre              8
director         199
writer           449
theater_date     359
dvd_date         359
currency        1220
box_office      1220
runtime           30
studio          1066
dtype: int64

### Rotten Tomatoes (Movie Information) | Data Understanding (Conclusion)

I have completed the exploratory data analysis of the tsv file associated with the Rotten Tomatoes dataset.  I placed the contents of the tsv file into a dataframe titled *df_rotten_movie*.  *Df_rotten_movie* has the following 12 columns - *id*, *synopsis*, *rating*, *genre*, *director*, *writer*, *theater_date*, *dvd_date*, *currency*, *box_office*, *runtime*, and *studio*.  *Df_rotten_movie* has a total of 1,560 rows.

## Rotten Tomatoes (Reviews) | Data Understanding

The second dataset for Rotten Tomatoes is also contained within a tsv file.  The data within the tsv file provides movie critic information.  I will utilize the following method - 'pd.read_csv()' - to examine the contents of the tsv file.

In [39]:
df_rotten_reviews = pd.read_csv('rt.reviews.tsv.gz', sep='\t', encoding='cp1252')

In [40]:
df_rotten_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [41]:
df_rotten_reviews.tail()

id                                             review rating   fresh  \
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic        publisher                date  
54427      Laura Sinagra           1    Village Voice  September 24, 2002  
54428  Michael Szymanski           0       Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0  EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0   Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0     Showbizz.net   November 12, 2002

In [42]:
df_rotten_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [43]:
df_rotten_reviews.shape

(54432, 8)

In [44]:
df_rotten_reviews_duplicates = df_rotten_reviews[df_rotten_reviews.duplicated()]
print(len(df_rotten_reviews_duplicates))

9


In [45]:
df_rotten_reviews.isna().sum()

id                0
review         5563
rating        13517
fresh             0
critic         2722
top_critic        0
publisher       309
date              0
dtype: int64

### Rotten Tomatoes (Reviews) | Data Understanding (Conclusion)

I have completed the exploratory data analysis of the tsv file associated with the second Rotten Tomatoes dataset.  I placed the contents of the tsv file into a dataframe titled *df_rotten_reviews*.  *Df_rotten_reviews* has the following 8 columns - *id*, *review*, *rating*, *fresh*, *critic*, *top_critic*, *publisher*, *date*.  *Df_rotten_reviews* has a total of 54,432 rows of data.

## The Movie DB | Data Understanding

The dataset for the Movie Database is contained within a csv file.  I will utilize the following method - 'pd.read_csv()' - to examine the contents of the csv file.  

In [46]:
# Placing the Movie DB file into a Dataframe

df_movie_db = pd.read_csv('tmdb.movies.csv.gz')

In [47]:
df_movie_db.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [48]:
df_movie_db.tail()

Unnamed: 0        genre_ids      id original_language  \
26512       26512         [27, 18]  488143                en   
26513       26513         [18, 53]  485975                en   
26514       26514     [14, 28, 12]  381231                en   
26515       26515  [10751, 12, 28]  366854                en   
26516       26516         [53, 27]  309885                en   

              original_title  popularity release_date                  title  \
26512  Laboratory Conditions         0.6   2018-10-13  Laboratory Conditions   
26513        _EXHIBIT_84xxx_         0.6   2018-05-01        _EXHIBIT_84xxx_   
26514           The Last One         0.6   2018-10-01           The Last One   
26515           Trailer Made         0.6   2018-06-22           Trailer Made   
26516             The Church         0.6   2018-10-05             The Church   

       vote_average  vote_count  
26512           0.0           1  
26513           0.0           1  
26514           0.0           1  
26515           0.0           1  
26516           0.0           1

In [49]:
df_movie_db.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


In [50]:
df_movie_db.shape

(26517, 10)

In [51]:
movie_db_duplicates = df_movie_db[df_movie_db.duplicated()]
print(len(movie_db_duplicates))

0


In [52]:
df_movie_db.isna().sum()

Unnamed: 0           0
genre_ids            0
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
dtype: int64

### The Movie DB | Data Understanding (Conclusion)

I have completed the exploratory data analysis of the csv file associated with the Movie Database dataset.  I placed the contents of the csv file into a dataframe titled *df_movie_db*.  *Df_movie_db* has the following 9 columns - *genre_ids*, *id*, *original_language*, *original_title*, *popularity*, *release_date*, *title*, *vote_average*, and *vote_count*.  *Df_movie_db* has a total of 26,517 rows of data.

## The Numbers | Data Understanding

The dataset for the Numbers is contained within a csv file.  I will utilize the the following method - 'pd.read_csv()' - to examine the contents of the csv file.  

In [53]:
df_numbers = pd.read_csv('tn.movie_budgets.csv.gz')

In [54]:
df_numbers.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [55]:
df_numbers.tail()

id  release_date                          movie production_budget  \
5777  78  Dec 31, 2018                         Red 11            $7,000   
5778  79   Apr 2, 1999                      Following            $6,000   
5779  80  Jul 13, 2005  Return to the Land of Wonders            $5,000   
5780  81  Sep 29, 2015           A Plague So Pleasant            $1,400   
5781  82   Aug 5, 2005              My Date With Drew            $1,100   

     domestic_gross worldwide_gross  
5777             $0              $0  
5778        $48,482        $240,495  
5779         $1,338          $1,338  
5780             $0              $0  
5781       $181,041        $181,041

In [56]:
df_numbers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [57]:
df_numbers.shape

(5782, 6)

In [58]:
numbers_duplicates = df_numbers[df_numbers.duplicated()]
print(len(numbers_duplicates))

0


In [59]:
# Determining whether or not there are any rows with null values

df_numbers.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

### The Numbers | Data Understanding (Conclusion)

I have completed the exploratory data analysis of the csv file associated with the Numbers dataset.  I placed the contents of the csv file into a dataframe titled *df_numbers*.  *Df_numbers* has the following 6 columns - *id*, *release_date*, *movie*, *production_budget*, *domestic_gross*, and *worldwide_gross*.  *Df_numbers* has a total of 5,782 rows of data.

## Data Understanding (Conclusion)

I completed my exploratory data analysis of the files from the following sources: IMDB, Box Office Mojo, Rotten Tomatos, the Movie DB, and the Numbers.  I want to combine the IMDB and the Numbers dataframes in order to acquire insights into the financial drivers of movies.  I can actually utilize the Box Office dataframe in lieu of The Numbers dataframe; however, I am using The Numbers dataframe for the following reasons:

- The Numbers dataframe contains more rows of data than Box Office Mojo (The Numbers dataset has 5,782 rows, while Box Office Mojo dataset has 3,387 rows)
- The Numbers dataframe contains financial data relatin to worldwide gross, and Box Office Mojo does not have this data

Before combining the IMDB and The Numbers dataframes, I will need to transform the Numbers dataframe.  Transformations will include the following:

- Converting the numerical data - in particular, the production budget, domestic gross, and worldwide gross data - from string data to float
- Creating a new column - *Net Profit* - by subtracting production budget from worldwide gross




# Data Preparation

## The Numbers Dataset Transformation

As part of the first phase of Data Preparation, I am going to transform the Numbers dataframe.  

- Converting the numerical data - in particular, the production budget, domestic gross, and worldwide gross data - from string data to float
- Creating a new column - *Net Profit* - by subtracting production budget from worldwide gross 

In [60]:
# Converting Production Budget from string to float

# Before converting Production Budget from string to float, the dollar signs and commas need to be removed

df_numbers['production_budget'] = df_numbers['production_budget'].str.replace('$', '').str.replace(',', '')


# Now, Production Budget can be converted from string to float

df_numbers['production_budget'] = pd.to_numeric(df_numbers['production_budget'])


In [61]:
# Converting Domestic Gross from string to float

# Before converting Domestic Gross from string to float, the dollar signs and commas need to be removed

df_numbers['domestic_gross'] = df_numbers['domestic_gross'].str.replace('$', '').str.replace(',', '')


# Now, Domestic Gross can be converted from string to float

df_numbers['domestic_gross'] = pd.to_numeric(df_numbers['domestic_gross'])


In [62]:
# Converting Worldwide Gross from string to float

# Before converting Worldwide Gross from string to float, the dollar signs and commas need to be removed

df_numbers['worldwide_gross'] = df_numbers['worldwide_gross'].str.replace('$', '').str.replace(',', '')

# Now, Worldwide Gross can be converted from string to float

df_numbers['worldwide_gross'] = pd.to_numeric(df_numbers['worldwide_gross'])


In [63]:
# Checking whether or not the columns - Production Budget, Domestic Gross, and Worldwide Gross - have been converted to float

df_numbers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   int64 
 4   domestic_gross     5782 non-null   int64 
 5   worldwide_gross    5782 non-null   int64 
dtypes: int64(4), object(2)
memory usage: 271.2+ KB


In [64]:
# Creating a column titled "Net Profit"

df_numbers['net_profit'] = df_numbers['worldwide_gross'] - df_numbers['production_budget']

In [65]:
# Checking to determine whether or not the new column - "New Profit" - has been created

df_numbers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   int64 
 4   domestic_gross     5782 non-null   int64 
 5   worldwide_gross    5782 non-null   int64 
 6   net_profit         5782 non-null   int64 
dtypes: int64(5), object(2)
memory usage: 316.3+ KB


In [66]:
# Examining the first five rows of the updated Numbers dataframe

df_numbers.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  net_profit  
0          425000000       760507625       2776345279  2351345279  
1          410600000       241063875       1045663875   635063875  
2          350000000        42762350        149762350  -200237650  
3          330600000       459005868       1403013963  1072413963  
4          317000000       620181382       1316721747   999721747

In [67]:
# Updating the following columns - "Production Budget", "Domestic Gross", "Worldwide Gross", and "Net Profit" to include "(dollars)"

df_numbers = df_numbers.rename(columns={'production_budget': 'production_budget (dollars)', 'domestic_gross': 'domestic_gross (dollars)', 'worldwide_gross': 'worldwide_gross (dollars)', 'net_profit': 'net_profit (dollars)'})

In [68]:
df_numbers.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget (dollars)  domestic_gross (dollars)  \
0                    425000000                 760507625   
1                    410600000                 241063875   
2                    350000000                  42762350   
3                    330600000                 459005868   
4                    317000000                 620181382   

   worldwide_gross (dollars)  net_profit (dollars)  
0                 2776345279            2351345279  
1                 1045663875             635063875  
2                  149762350            -200237650  
3                 1403013963            1072413963  
4                 1316721747             999721747

## Merging IMDB with The Numbers

In [69]:
# Examining IMDB dataframe again with ".info()" method

df_imdb_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [70]:
# Examining IMDB dataframe again with ".shape"

df_imdb_movie_basics.shape

(146144, 6)

In [71]:
# Examining The Numbers dataframe again with ".info()" method

df_numbers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 7 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   id                           5782 non-null   int64 
 1   release_date                 5782 non-null   object
 2   movie                        5782 non-null   object
 3   production_budget (dollars)  5782 non-null   int64 
 4   domestic_gross (dollars)     5782 non-null   int64 
 5   worldwide_gross (dollars)    5782 non-null   int64 
 6   net_profit (dollars)         5782 non-null   int64 
dtypes: int64(5), object(2)
memory usage: 316.3+ KB


In [72]:
# Examining The Numbers dataframe again with ".shape" 

df_numbers.shape

(5782, 7)

As a reminder, I want to perform the left join of both dataframes - "df_numbers" and "df_imdb_movie_basics" - with df_numbers as the left table.  I want to use the *movie* column from the df_numbers dataframe and the *primary title* column from the df_imdb_movie_basics in order to perform the left join.  Before performing the left join, I am going to check for duplicates in the aforementioned columns. 

In [73]:
# Checking for duplicates in the *movie* column of the df_numbers dataframe

df_numbers['movie'].duplicated().sum()

84

There is a sum of 84 duplicates for the *movie* column in the df_numbers dataframe.  I am going to examine the duplicates in the respective dataframe.

In [83]:
df_numbers[df_numbers['movie'].duplicated(keep=False)][0:50]

id  release_date                          movie  \
26     27   May 4, 2012                   The Avengers   
38     39  May 14, 2010                     Robin Hood   
39     40  Dec 14, 2005                      King Kong   
50     51   Mar 5, 2010            Alice in Wonderland   
64     65   Jun 9, 2017                      The Mummy   
69     70  May 10, 2013               The Great Gatsby   
80     81  May 24, 2019                        Aladdin   
96     97  Apr 15, 2016                The Jungle Book   
134    35  Mar 17, 2017           Beauty and the Beast   
140    41  May 16, 2014                       Godzilla   
203     4  Jul 15, 2016                   Ghostbusters   
243    44  Mar 27, 2015                           Home   
267    68   Aug 8, 2014   Teenage Mutant Ninja Turtles   
271    72   Apr 1, 2010            Clash of the Titans   
273    74  May 19, 1998                       Godzilla   
278    79   Aug 3, 2012                   Total Recall   
295    96  Feb 12, 2014                        RoboCop   
297    98   Aug 7, 2015                 Fantastic Four   
299   100  Jul 22, 2005                     The Island   
302     3   Oct 5, 2018                          Venom   
340    41  Jun 16, 2004    Around the World in 80 Days   
359    60  Nov 17, 2006                  Casino Royale   
366    67  Jul 27, 2001             Planet of the Apes   
383    84  Jul 25, 2014                       Hercules   
392    93  Dec 25, 2003                      Peter Pan   
399   100  Dec 25, 2015                    Point Break   
408     9  Nov 21, 2018                     Robin Hood   
412    13  Mar 13, 2015                     Cinderella   
422    23  Aug 19, 2016                        Ben-Hur   
434    35   Apr 9, 2004                      The Alamo   
475    76  Aug 19, 2011            Conan the Barbarian   
484    85   Jul 8, 2005                 Fantastic Four   
543    44   May 7, 1999                      The Mummy   
571    72  Dec 12, 2008  The Day the Earth Stood Still   
650    51  Apr 16, 1999                           Life   
707     8  Jun 13, 1997                       Hercules   
779    80   Jun 1, 1990                   Total Recall   
790    91  Aug 12, 2016                Pete’s Dragon   
883    84   Apr 2, 2004                        Hellboy   
934    35  Aug 14, 1998                   The Avengers   
965    66  Mar 24, 2017                           Life   
1046   47  Jul 22, 1998                         Lolita   
1055   56  Jun 16, 2000                          Shaft   
1176   77  Apr 12, 2019                        Hellboy   
1323   24  Dec 25, 2002                      Pinocchio   
1376   77  Jun 11, 2010                 The Karate Kid   
1420   21  Feb 18, 2011                        Unknown   
1549   50  Sep 30, 2005                       Serenity   
1588   89  Nov 21, 2008                       Twilight   
1648   49  Apr 30, 2010      A Nightmare on Elm Street   

      production_budget (dollars)  domestic_gross (dollars)  \
26                      225000000                 623279547   
38                      210000000                 105487148   
39                      207000000                 218080025   
50                      200000000                 334191110   
64                      195000000                  80101125   
69                      190000000                 144840419   
80                      182000000                 246734314   
96                      175000000                 364001123   
134                     160000000                 504014165   
140                     160000000                 200676069   
203                     144000000                 128350574   
243                     130000000                 177397510   
267                     125000000                 191204754   
271                     125000000                 163214888   
273                     125000000                 136314294   
278                     125000000       

After further reviewing the duplicates in the *movie* column in the df_numbers dataframe, I noticed that there are movies with the same title, but they have different release dates.  In reality, they are actually different movies.  For example, *Robin Hood* has release dates of May 14th, 2010, and November 21st, 2018.  These are different movies because Robin Hood has been created multiple times within Hollywood.

In [74]:
df_imdb_movie_basics['primary_title'].duplicated().sum()

10073

There is a sum of 10,073 duplicates for the *primary title* column in the df_imdb_movie_basics dataframe.  I am going to examine the duplicates in the respective dataframe.

In [90]:
df_imdb_movie_basics[df_imdb_movie_basics['primary_title'].duplicated(keep=False)][0:60]

movie_id                     primary_title  \
6    tt0112502                           Bigfoot   
19   tt0247643  Los pájaros se van con la muerte   
21   tt0250404                         Godfather   
30   tt0285252                    Life's a Beach   
34   tt0297400                         Snowblind   
35   tt0306058                     Second Coming   
40   tt0326592                     The Overnight   
42   tt0328810                           My Land   
48   tt0337692                       On the Road   
51   tt0339736                   The Evil Within   
61   tt0372538                               Spy   
67   tt0383010                 The Three Stooges   
74   tt0402887                  Capture the Flag   
85   tt0419692                        Disconnect   
115  tt0443272                           Lincoln   
117  tt0443465                      Before We Go   
120  tt0445054                            7 Days   
128  tt0451108                         Mad World   
129  tt0451154                       The Passing   
131  tt0451279                      Wonder Woman   
133  tt0452664                    Party Crashers   
177  tt0473088                              Stan   
185  tt0477080                       Unstoppable   
188  tt0478304                  The Tree of Life   
192  tt0479204                      Perfect Life   
196  tt0480255                        The Losers   
200  tt0482459                           Bandage   
211  tt0490075                         Aftermath   
213  tt0490215                           Silence   
214  tt0490236                       The Invited   
220  tt0493076                              Nina   
224  tt0495024                         Bad Blood   
229  tt0499097                   Without Remorse   
236  tt0765453                 Have You Seen Me?   
252  tt0775489                   The Illusionist   
254  tt0780504                             Drive   
272  tt0790788              The Yellow Wallpaper   
275  tt0798817                                13   
276  tt0800054                     The Guardians   
280  tt0800353                           Nemesis   
282  tt0803061                           Red Dog   
287  tt0804509                          The Mule   
288  tt0805617                             Spoon   
292  tt0808510                       Tooth Fairy   
305  tt0815101                             Ashes   
313  tt0817177                           Flipped   
324  tt0826823                             Magic   
325  tt0828135               Wild in the Streets   
330  tt0831881                              Land   
341  tt0840361                          The Town   
351  tt0847180                              Loot   
358  tt0848554                      The Sighting   
363  tt0850709                             Risen   
368  tt0853002                     A Summer Song   
383  tt0865557                        Letting Go   
385  tt0865911                         Catharsis   
396  tt0875613                              Rift   
404  tt0879859                              Torn   
411  tt0884325                        Drain Baby   
425  tt0893310                              Blur   

                       original_title  start_year  runtime_minutes  \
6                             Bigfoot        2017              NaN   
19   Los pájaros se van con la muerte        2011            110.0   
21                          Godfather        2012              NaN   
30                     Life's a Beach        2012            100.0   
34                          Snowblind        2015              NaN   
35                      Second Coming        2012             95.0   
40                      The Overnight        2010             88.0   
42                      Aamaar Bhuvan        2018            102.0   
48                        On the Road        2012            124.0   
51                    The Evil Within        2017             98.0   
61                                Spy        2011            11

After running the following code - 'df_imdb_movie_basics[df_imdb_movie_basics['primary_title'].duplicated(keep=False)][0:60]' -  I still do not know why there are duplicates. However, I am only interested in the data for the following column - df_imdb_movie_basics['runtime_minutes']. There are null values for the column - df_imdb_movie_basics['runtime_minutes']. As a result, I am going to delete the rows with these null values from the dataframe.

In [95]:
# Eliminating the rows of data for null values in the "runtime_minutes" column of the df_imdb_movie_basics dataframe

df_imdb_movie_basics = df_imdb_movie_basics.dropna(subset=['runtime_minutes'])

In [96]:
# Checking for duplicates in the "primary title' column of the df_imdb_movie_basics dataframe

df_imdb_movie_basics['primary_title'].duplicated().sum()

6946

Eliminating the rows with null values in the 'runtime_minutes' column reduced the number of duplicates from 10,073 to 6,946.  However, duplicates still persist in the df_imdb_movie_basics dataframe.  Instead of only using the movie column from the df_numbers dataframe and the primary title column from the df_imdb_movie_basics for the left join, I will enhance the left join by using an additional column that references the release year from both dataframes. 

In [97]:
# Converting the 'release_date' column of the df_numbers dataframe to datetime format

df_numbers['release_date'] = pd.to_datetime(df_numbers['release_date'], errors='coerce')

In [99]:
# Creating a new column - 'release_year' in the df_numbers dataframe

df_numbers['release_year'] = df_numbers['release_date'].dt.year

In [100]:
# Checking to determine whether or not the 'release_year' column was created

df_numbers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   id                           5782 non-null   int64         
 1   release_date                 5782 non-null   datetime64[ns]
 2   movie                        5782 non-null   object        
 3   production_budget (dollars)  5782 non-null   int64         
 4   domestic_gross (dollars)     5782 non-null   int64         
 5   worldwide_gross (dollars)    5782 non-null   int64         
 6   net_profit (dollars)         5782 non-null   int64         
 7   release_year                 5782 non-null   int64         
dtypes: datetime64[ns](1), int64(6), object(1)
memory usage: 361.5+ KB


In [103]:
# Checking for null values in the 'release_year' column

df_numbers.isna().sum()

id                             0
release_date                   0
movie                          0
production_budget (dollars)    0
domestic_gross (dollars)       0
worldwide_gross (dollars)      0
net_profit (dollars)           0
release_year                   0
dtype: int64

In [104]:
# Performing the left join of both dataframes - "df_numbers" and "df_imdb_movie_basics" - with df_numbers as the left table

# df_imdb_numbers = df_numbers.merge(df_imdb_movie_basics, how='left', left_on='column_in_df_numbers', right_on='column_in_df_imdb_movie_basics')

df_imdb_numbers = df_numbers.merge(df_imdb_movie_basics, how='left', left_on=['movie', 'release_year'], right_on=['primary_title', 'start_year'])

In [105]:
# Examining the new dataframe - 'df_imdb_numbers'

df_imdb_numbers.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5833 entries, 0 to 5832
Data columns (total 14 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   id                           5833 non-null   int64         
 1   release_date                 5833 non-null   datetime64[ns]
 2   movie                        5833 non-null   object        
 3   production_budget (dollars)  5833 non-null   int64         
 4   domestic_gross (dollars)     5833 non-null   int64         
 5   worldwide_gross (dollars)    5833 non-null   int64         
 6   net_profit (dollars)         5833 non-null   int64         
 7   release_year                 5833 non-null   int64         
 8   movie_id                     1521 non-null   object        
 9   primary_title                1521 non-null   object        
 10  original_title               1521 non-null   object        
 11  start_year                   1521 non-null 

In [112]:
# Checking for duplicate rows in the new dataframe - 'df_imdb_numbers'

imdb_numbers_duplicates = df_imdb_numbers[df_imdb_numbers.duplicated()]
print(len(imdb_numbers_duplicates))

0


In [111]:
# Examining a sample of the rows in the new dataframe - 'df_imdb_numbers'

df_imdb_numbers.tail(20)

id release_date                            movie  \
5813  63   1997-04-11                   Pink Flamingos   
5814  64   2006-04-28         Grip: A Criminal's Story   
5815  65   2007-12-31                      Tin Can Man   
5816  66   2001-03-09                          Dayereh   
5817  67   2006-04-28                            Clean   
5818  68   2001-07-06                             Cure   
5819  69   2004-05-28                   On the Downlow   
5820  70   1996-04-01                             Bang   
5821  71   2008-08-14  The Rise and Fall of Miss Thang   
5822  72   2015-05-19                 Family Motocross   
5823  73   2012-01-13                        Newlyweds   
5824  74   1993-02-26                      El Mariachi   
5825  75   2004-10-08                           Primer   
5826  76   2006-05-26                           Cavite   
5827  77   2004-12-31                  The Mongol King   
5828  78   2018-12-31                           Red 11   
5829  79   1999-04-02                        Following   
5830  80   2005-07-13    Return to the Land of Wonders   
5831  81   2015-09-29             A Plague So Pleasant   
5832  82   2005-08-05                My Date With Drew   

      production_budget (dollars)  domestic_gross (dollars)  \
5813                        12000                    413802   
5814                        12000                      1336   
5815                        12000                         0   
5816                        10000                    673780   
5817                        10000                    138711   
5818                        10000                     94596   
5819                        10000                      1987   
5820                        10000                       527   
5821                        10000                       401   
5822                        10000                         0   
5823                         9000                      4584   
5824                         7000                   2040920   
5825                         7000                    424760   
5826                         7000                     70071   
5827                         7000                       900   
5828                         7000                         0   
5829                         6000                     48482   
5830                         5000                      1338   
5831                         1400                         0   
5832                         1100                    181041   

      worldwide_gross (dollars)  net_profit (dollars)  release_year movie_id  \
5813                     413802                401802          1997      NaN   
5814                       1336                -10664          2006      NaN   
5815                          0                -12000          2007      NaN   
5816                     673780                663780          2001      NaN   
5817                     138711                128711          2006      NaN   
5818                      94596                 84596          2001      NaN   
5819                       1987                 -8013          2004      NaN   
5820                        527                 -9473          1996      NaN   
5821                        401                 -9599          2008      NaN   
5822                          0                -10000          2015      NaN   
5823                       4584                 -4416          2012      NaN   
5824                    2041928               2034928          1993      NaN   
5825                     841926                834926          2004      NaN   
5826                      71644                 64644          2006      NaN   
5827                        900                 -6100          2004      NaN   
5828                          0                 -7000          2018      NaN   
5829                     240495                234495          1999      NaN   
5830                   

In [109]:
# Examining for null values in the new dataframe - 'df_imdb_numbers'

df_imdb_numbers.isna().sum()

id                                0
release_date                      0
movie                             0
production_budget (dollars)       0
domestic_gross (dollars)          0
worldwide_gross (dollars)         0
net_profit (dollars)              0
release_year                      0
movie_id                       4312
primary_title                  4312
original_title                 4312
start_year                     4312
runtime_minutes                4312
genres                         4314
dtype: int64

Why so much data was cut from the new dataframe - df_imdb_numbers?

- Difference in the amounts of data in the original dataframes.
- Both of the dataframes had different movies listed

# Modeling

# Overall Conclusion and Recommendations

# Next Steps